promql/parse.go

```diff
@@ -245,41 +245,41 @@ func (p *parser) typecheck(node Node) (err error) {
 
 // next returns the next token.
 func (p *parser) next() item {
-	if p.peekCount > 0 {
-		p.peekCount--
+	if p.peeking {
+		p.peeking = false
```

Comment:
You could unconditionally set this to false

Reply:
True, thanks!